### PR TITLE
Support for annotation based collection point 

### PR DESCRIPTION
**Pull request by hiteshbedre**

### CODE REVIEW

File: src/main/scala/ai/privado/exporter/CollectionExporter.scala [hunk 1]
```diff
@@ -77,13 +78,38 @@ class CollectionExporter(cpg: Cpg) {
...
+    })
+
     val ruleInfo = ExporterUtility.getRuleInfoForExporting(collectionId)
     CollectionModel(
       collectionId,
       ruleInfo.name,
       ruleInfo.isSensitive,
       collectionParameterMapById
         .map(entrySet => processByParameterId(entrySet._1, entrySet._2.toList))
+        .toList ::: collectionLocalMapById
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/CollectionExporter.scala:111]: what does the `:::` operator does?

File: src/main/scala/ai/privado/exporter/CollectionExporter.scala [hunk 2]
```diff
@@ -77,13 +78,38 @@ class CollectionExporter(cpg: Cpg) {
         })
     })
 
+    collectionMethods.foreach(collectionMethod => {
+      collectionMethod.local
+        .and(_.tag.nameExact(Constants.id))
+        .foreach(localVar => {
+          try {
+            def addToMap(parameterId: String): Unit = {
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/CollectionExporter.scala:86]: rename the variable `parameterId` to `localId` or `localVariableId`

File: src/main/scala/ai/privado/exporter/CollectionExporter.scala (hunk 1 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/CollectionExporter.scala:111]: Its used for list concatenation.
Reference: https://alvinalexander.com/scala/how-merge-scala-lists-concatenate/

File: src/main/scala/ai/privado/exporter/CollectionExporter.scala (hunk 2 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/CollectionExporter.scala:86]: Changes done here: https://github.com/Privado-Inc/privado-core/pull/370/commits/a2b4c1f93171c273bb23686474e8db69c4ae3235

Comment (khemrajrathore) [reply src/main/scala/ai/privado/exporter/CollectionExporter.scala:86]: @hiteshbedre Looks like you updated the wrong function, update it on line number 86 function

File: src/main/scala/ai/privado/exporter/CollectionExporter.scala [hunk 3]
```diff
@@ -115,6 +141,33 @@ class CollectionExporter(cpg: Cpg) {
     )
   }
 
+  def processByLocalVariableId(
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/CollectionExporter.scala:144]: @hiteshbedre We don't need this additional function and also `getCollectionUrlForLocalVar` the existing one can be used by a minor update

File: src/main/scala/ai/privado/exporter/JSONExporter.scala [hunk 4]
```diff
@@ -64,6 +63,7 @@ import ExecutionContext.Implicits.global
 import duration._
 import scala.language.postfixOps
 import scala.util.{Failure, Success}
+import io.shiftleft.semanticcpg.language._
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/JSONExporter.scala:66]: remove the unused import

File: src/main/scala/ai/privado/exporter/JSONExporter.scala [hunk 5]
```diff
@@ -34,15 +34,14 @@ import ai.privado.model.exporter.{
   SourceModel,
   SourceProcessingModel
 }
-import ai.privado.model.{Constants, InternalTag, PolicyThreatType}
+import ai.privado.model.{CatLevelOne, Constants, DataFlowPathModel, InternalTag, PolicyThreatType}
```
Comment (khemrajrathore) [inline src/main/scala/ai/privado/exporter/JSONExporter.scala:37]: remove the unused import

File: src/main/scala/ai/privado/exporter/CollectionExporter.scala (hunk 2 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/CollectionExporter.scala]: Corrected here: https://github.com/Privado-Inc/privado-core/pull/370/commits/62c970a3cf8860ab01497b4cc82b3e7e2e36d41e

File: src/main/scala/ai/privado/exporter/JSONExporter.scala (hunk 4 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/JSONExporter.scala:66]: This is used for IDE to resolve some implicit function. Not related to any compile time or runtime time dependency.

File: src/main/scala/ai/privado/exporter/JSONExporter.scala (hunk 5 above)
Comment (hiteshbedre) [reply src/main/scala/ai/privado/exporter/JSONExporter.scala]: Removed here: https://github.com/Privado-Inc/privado-core/pull/370/commits/ac73a384505ebd99f64ef649231a9b12f8e0a46a

**Review by khemrajrathore — DISMISSED** [review]
LGTM

**Comment by khemrajrathore** [comment]
@hiteshbedre Change the base branch to python-experimental

**Pull request merged**